lisa/tools/qemu.py

```diff
@@ -122,14 +122,27 @@ def create_vm(
             self.delete_vm()
 
         # start qemu
-        self.run(
+        result = self.run(
             cmd,
             sudo=True,
             shell=True,
-            expected_exit_code=0,
-            expected_exit_code_failure_message=f"Unable to start VM {guest_image_path}",
         )
 
+        if result.exit_code != 0:
+            if "ret == cpu->kvm_msr_buf->nmsrs" in result.stdout:
+                # Known issue with qemu on AMD
+                # https://bugs.launchpad.net/qemu/+bug/1661386
+                raise LisaException(
+                    "Unable to start VM. "
+                    "Found `ret == cpu->kvm_msr_buf->nmsrs` in stdout. "
+                    "Known issue with qemu on AMD on older kernels"
+                )
+            elif "Cannot allocate memory" in result.stdout:
+                raise SkippedException(
```

Comment:
Can you share more details about this output? Is it caused by the system memory not enough?

Reply:
```
2023-04-03 02:47:17.262[4164][DEBUG] lisa.env[generated_0].node[0].cmd[3058] cmd: ['sudo', 'sh', '-c', '/usr/libexec/qemu-kvm -cpu host -smp 2 -m 4096 -hda /mnt/image.qcow2 -device e1000,netdev=net0,mac=72:82:e9:8a:5f:a0 -netdev user,id=net0,hostfwd=tcp::60024-:22 -drive id=datadisk-0,file=/dev/sdc,cache=none,if=none,format=raw,aio=threads -device virtio-scsi-pci -device scsi-hd,drive=datadisk-0 -enable-kvm -display none -daemonize '], cwd: None, shell: True, sudo: True, nohup: False, posix: True, remote: True
2023-04-03 02:47:18.121[7240][DEBUG] lisa.env[generated_0].node[0].cmd[3058].stdout qemu-kvm: cannot set up guest memory 'pc.ram': Cannot allocate memory
2023-04-03 02:47:18.137[4164][DEBUG] lisa.env[generated_0].node[0].cmd[3058] execution time: 0.874 sec, exit code: 1
```


These were running on a system with 2GB of memory